src/databricks/labs/remorph/transpiler/transpile_status.py

```diff
@@ -1,30 +1,31 @@
+import abc
 from dataclasses import dataclass
 from pathlib import Path
 
 
 @dataclass
-class _BaseError:
+class TranspileError(abc.ABC):
     file_path: Path
-    exception: str
+    error_msg: str
 
     def __str__(self):
-        return f"{type(self).__name__}(file_path='{self.file_path!s}', exception='{self.exception}')"
+        return f"{type(self).__name__}(file_path='{self.file_path!s}', error_msg='{self.error_msg}')"
 
 
 @dataclass
-class ParserError(_BaseError):
+class ParserError(TranspileError):
     pass
 
 
 @dataclass
-class ValidationError(_BaseError):
+class ValidationError(TranspileError):
     pass
 
 
 @dataclass
 class TranspileStatus:
     file_list: list[Path]
-    no_of_queries: int
+    no_of_transpiled_queries: int
```

Comment:
Let's clarify. Should it be:
 - the number of files (deterministic, but can be inferred for file_list)
 - the number of source queries (non-deterministic since we don't know how many queries were not successfully parsed)
 - the number of transpiled queries  (deterministic)
I picked the latter because the number of source queries can be inferred from no_of_transpiled_queries + parse_error_count + validate_error_count ( adding numbers is simpler that subtracting them 😆 )